### PR TITLE
[SPARK-14897] [SQL] upgrade to jetty 9.2.16

### PR DESCRIPTION
**Pull request by bomeng**
## What changes were proposed in this pull request?

Since Jetty 8 is EOL (end of life) and has critical security issue [http://www.securityweek.com/critical-vulnerability-found-jetty-web-server], I think upgrading to 9 is necessary. I am using latest 9.2 since 9.3 requires Java 8+.

`javax.servlet` and `derby` were also upgraded since Jetty 9.2 needs corresponding version.
## How was this patch tested?

Manual test and current test cases should cover it.

### CODE REVIEW

**Comment by srowen** [comment]
Looking pretty good to me. It needs a rebase, and I think the test failure is unrelated.

**Comment by bomeng** [comment]
The test failure was caused by timeout for HiveThriftHttpServerSuite and SingleSessionSuite... have not figure out the cause, any suggestion will be welcome.

**Comment by srowen** [comment]
Hm, that could be a legitimate failure, since failure to start an HTTP server here sounds suspiciously linked. Looking at the test it does collect output from the server process but doesn't seem to dump it as expected in this output. Does `HiveThriftServer2Suites` pass for you locally? if not maybe you can nudge the code to figure out why it's not dumping the error, or see why the server fails to start.

**Comment by bomeng** [comment]
retest please

**Comment by bomeng** [comment]
@srowen Finally I've got it working. Servlet and Derby were upgraded as well due to requirement of Jetty. Please review.

File: core/pom.xml [hunk 1]
```diff
@@ -125,12 +125,17 @@
       <artifactId>jetty-servlet</artifactId>
       <scope>compile</scope>
     </dependency>
+    <dependency>
+      <groupId>org.eclipse.jetty</groupId>
+      <artifactId>jetty-servlets</artifactId>
+      <scope>compile</scope>
+    </dependency>
     <!-- Because we mark jetty as provided and shade it, its dependency
```
Comment (holdenk) [inline core/pom.xml]: I think this comment needs to be updated (you aren't including orbit here anymore yes?)

**Comment by srowen** [comment]
@bomeng this is looking nice. The last worry I have is about the javax.servlet dependency. I remember we had significant problems with servlet containers bringing in their own copy under their own artifact name, and conflicting with the main `javax.servlet` artifact. I think we ended up excluding everything except Jetty's copy. That seems to be changed in this PR, which could be fine. But if you would try `mvn dependency:tree` and grep for any "javax.+servlet" artifacts, and see if anything whose version looks like it's less than 3.1.0 is still included. (Likewise maybe we can remove some old excludes of stuff like this, but I'm not as worried.)

If there are no other servlet APIs coming in, I'm down with this.

**Comment by srowen** [comment]
Jenkins retest this please

**Comment by srowen** [comment]
Jenkins retest this please

File: core/pom.xml (hunk 1 above)
Comment (bomeng) [inline core/pom.xml]: We are actually just using orbit javax.servlet package, so upgrade to javax.servlet is ok. But not sure what I should write for the comment.

**Comment by bomeng** [comment]
@srowen sorry for the late reply, I did not notice it. I have run the mvn dependency:tree and only javax.servlet-api 3.1.0 is listed, so it should be fine.

File: core/pom.xml (hunk 1 above)
Comment (srowen) [inline core/pom.xml]: You can remove the comment. You say you see just one version javax.servlet, which is good, but do you now see org.eclipse.jetty.orbit:javax.servlet in the dependency tree? I'm guessing not, but just want to make sure that's what you're looking for -- any possible copy of the servlet API from another source.

Comment (bomeng) [inline core/pom.xml]: Just tried, it returns nothing. I will remove the comment shortly.

File: yarn/pom.xml [hunk 2]
```diff
@@ -102,6 +102,10 @@
       <groupId>org.eclipse.jetty</groupId>
       <artifactId>jetty-servlet</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.eclipse.jetty</groupId>
+      <artifactId>jetty-servlets</artifactId>
+    </dependency>
```
Comment (srowen) [inline yarn/pom.xml]: Looks good to me. Just confirming there really are both "jetty-servlet" and "jetty-servlets" artifacts that we need both of?

Comment (bomeng) [inline yarn/pom.xml]: yes, we need both, they have different content.

**Comment by srowen** [comment]
Merged to master/2.0

**Pull request closed without merging**